src/go/rpk/pkg/config/schema.go

```diff
@@ -85,6 +85,7 @@ type NamedSocketAddress struct {
 }
 
 type TLS struct {
+	Enabled        bool   `yaml:"enabled,omitempty" mapstructure:"enabled,omitempty" json:"enabled"`
```

Comment:
Is this field necessary? It is not used anywhere in this PR (unless there is a collapsed file ...). I thought the existence / non-nil-ness of this struct is what would signify enabling TLS without specifying overriding files.

Reply:
Indeed it's not. I removed it.